qtfred/src/mission/dialogs/ReinforcementsEditorDialogModel.cpp

```diff
@@ -0,0 +1,316 @@
+#include "ReinforcementsEditorDialogModel.h"
+#include "ship/ship.h"
+
+namespace fso {
+namespace fred {
+namespace dialogs {
+
+ReinforcementsDialogModel::ReinforcementsDialogModel(QObject* parent, EditorViewport* viewport)
+	: AbstractDialogModel(parent, viewport)
+{
+	initializeData();
+}
+
+void ReinforcementsDialogModel::initializeData()
+{
+	for (int i = 0; i < Num_reinforcements; i++) {
+		_reinforcementList.emplace_back(Reinforcements[i].name, Reinforcements[i].uses, Reinforcements[i].arrival_delay);
+	}
+
+	// add the wings to the model's internal storage
+	for (auto& currentWing : Wings) {
+
+		if (currentWing.wave_count == 0) {
+			continue;
+		}
+
+		// compiler seems to be optimizing this out despite it being a perfectly valid check.
+		for (auto& playerTest : currentWing.ship_index) {
+			// no players allowed in reinforcements!
+			if (playerTest > -1 && (Objects[Ships[playerTest].objnum].type == OBJ_START)){
+				continue;
+			}
+		}
```

Comment:
This `continue` is only continuing the innermost loop.  Since it does nothing else, that's why the compiler optimizes it away.  You would need to either use `goto`, or set a flag, break, check the flag, and *then* `continue`.